frame/vesting/src/lib.rs

```diff
@@ -131,6 +139,44 @@ pub mod pallet {
 
 		/// Weight information for extrinsics in this pallet.
 		type WeightInfo: WeightInfo;
+
+		/// Maximum number of vesting schedules an account may have at a given moment.
+		#[pallet::constant]
+		type MaxVestingSchedules: Get<u32>;
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		#[cfg(feature = "try-runtime")]
+		fn pre_upgrade() -> Result<(), &'static str> {
+			migrations::v1::pre_migrate::<T>()
+		}
+
+		fn on_runtime_upgrade() -> Weight {
+			if StorageVersion::<T>::get() == Releases::V0 {
+				StorageVersion::<T>::put(Releases::V1);
+				migrations::v1::migrate::<T>()
+			} else {
+				T::DbWeight::get().reads(1)
+			}
+		}
+
+		#[cfg(feature = "try-runtime")]
+		fn post_upgrade() -> Result<(), &'static str> {
+			migrations::v1::post_migrate::<T>()
+		}
+
+		fn integrity_test() {
+			sp_std::if_std! {
+				sp_io::TestExternalities::new_empty().execute_with(||
```

Comment:
not sure if the externalities is needed here. All of these (`MinVestedTransfer ` etc.) are constant, not storage items

Reply:
If I move them out of externalities I get this error in polkadot test runtime:

```
running 3 tests
test __pallet_staking_reward_curve_test_module::reward_curve_piece_count ... ok
test __construct_runtime_integrity_test::runtime_integrity_tests ... FAILED
test __pallet_staking_reward_curve_test_module::reward_curve_precision ... ok

failures:

---- __construct_runtime_integrity_test::runtime_integrity_tests stdout ----
thread '__construct_runtime_integrity_test::runtime_integrity_tests' panicked at '`get_version_1` called outside of an Externalities-provided environment.', /home/emostov/remote-builds/substrate-vesting/primitives/io/src/lib.rs:96:5
stack backtrace:
   0: rust_begin_unwind
   ```

Reply:
I think its `<T as Config>::Currency::minimum_balance()`

Reply:
yup, then it is okay.

Reply:
@kianenigma lmk if this seems alright: dcdcda5697ab7d6de28584507f0cc632aad1d709

Reply:
ah didn't see that reply thanks!